studio/components/grid/components/header/filter/FilterPopover.tsx

```diff
@@ -48,7 +48,7 @@ const Filter: FC = () => {
     <div className="space-y-2 py-2">
       <div className="space-y-2">
         {state.filters.map((_, index) => (
```

Comment:
should not use array index for identifying the row for rerenders. can replace with a string based on filter value.